service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/OrganizationToAgentMapperTest.java

```diff
@@ -51,16 +56,22 @@ public void When_MappingOrganization_Expect_AgentResourceXml(String inputJson, S
         Organization organization = new FhirParseService().parseResource(jsonInput, Organization.class);
         var outputMessage = organizationToAgentMapper.mapOrganizationToAgent(organization);
         assertThat(outputMessage)
-            .withFailMessage(TestArgumentsLoaderUtil.FAIL_MESSAGE, inputJson, outputXml)
+            .describedAs(TestArgumentsLoaderUtil.FAIL_MESSAGE, inputJson, outputXml)
             .isEqualToIgnoringWhitespace(expectedOutput);
     }
 
+    @Test
+    public void When_MappingOrganizationInner_Expect_AgentResourceXmlInner() throws IOException {
+        var jsonInput = ResourceTestFileUtils.getFileContent(ORGANIZATION_FOR_INNER_MAPPER_JSON);
+        var expectedOutput = ResourceTestFileUtils.getFileContent(ORGANIZATION_FOR_INNER_MAPPER_XML);
+
+        Organization organization = new FhirParseService().parseResource(jsonInput, Organization.class);
+        var outputMessage = organizationToAgentMapper.mapOrganizationToAgentInner(organization);
+        assertThat(outputMessage).isEqualToIgnoringWhitespace(expectedOutput);
+    }
+
     private static Stream<Arguments> readTestCases() {
         return TestArgumentsLoaderUtil.readTestCases(ORGANIZATION_FILE_LOCATION);
     }
 
-    @AfterEach
```

Comment:
message context has to be cleared after each test to avoid random errors with cached ids.